In [1]:
using MPSGE

In [2]:
m = Model()

@parameter(m, diff, 0.0)
@parameter(m, sub_elas_a, 1.5)
@parameter(m, sub_elas_b, 2.)
@parameter(m, sub_elas_w, .5)
# @parameter(m, sub_elas_a, 1.)
# @parameter(m, sub_elas_b, 1.)
# @parameter(m, sub_elas_w, 1.)
@parameter(m, t_elas_a, 0.0)
@parameter(m, t_elas_b, 0.0)

@sector(m, A)
@sector(m, B)
@sector(m, W)

@commodity(m, PX)
@commodity(m, PY)
@commodity(m, PW)
@commodity(m, PL)
@commodity(m, PK)

@consumer(m, CONS, benchmark=200.0)

# @production(m, A, 0, 1, [Output(PX, 80),          Output(PY, 20)], [Input(PL, 40), Input(PK, 60)])
@production(m, A, :($t_elas_a*1.), :($sub_elas_a*1.), [Output(PX, 80),          Output(PY, 20)], [Input(PL, 40), Input(PK, 60)])
# @production(m, B, 0, 1, [Output(PX, 20), Output(PY, 80)], [Input(PL, 60), Input(PK, 40)])
@production(m, B, :($t_elas_b*1.), :($sub_elas_b*1.), [Output(PX, 20), Output(PY, 80)], [Input(PL, 60), Input(PK, 40)])
# @production(m, B, :($t_elas_b*1.), :($sub_elas_b*1.), [Output(PX, :(20+$diff)), Output(PY, 80)], [Input(PL, 60), Input(PK, 40)])
@production(m, W, 0, :($sub_elas_w*1.), [Output(PW, 200.0)],[Input(PX, :(100.0+$diff)), Input(PY, 100.0)])
# @production(m, W, 0, :($var"var"var"var"sub_elas_w""""*1.), [Output(PW, 200.0)], [Input(PX, 100.0), Input(PY, 100.0)])
# @production(m, W, 0.,1, [Output(PW, :(200.0+$diff))], [Input(PX, :(100.0+$diff)), Input(PY, 100.0)])

@demand(m, CONS, 1., [Demand(PW, 200.)], [Endowment(PL, 100.0), Endowment(PK, 100.0)])

solve!(m, cumulative_iteration_limit=0)

# set_value(diff, 10.0)
set_fixed!(PW, true)
set_fixed!(CONS, true)
solve!(m)


PL†A: 40.0


PK†A: 60.0
PL†B: 60.0


PK†B: 40.0
PX†W: 

100.0 + 0.0
PY†W: 100.0


Out: s.elas1.5 * 1.0PX‡A: 80.0


Out: s.elas1.5 * 1.0PY‡A: 20.0
Out: s.elas2.0 * 1.0PX‡B: 20.0
Out: s.elas2.0 * 1.0PY‡B: 80.0


Out: s.elas0.5 * 1.0PW‡W: 200.0


Reading options file C:\Users\Eli\AppData\Local\Temp\jl_38C6.tmp
 > cumulative_iteration_limit 0
Read of options file complete.

Path 5.0.03 (Fri Jun 26 10:05:33 2020)
Written by Todd Munson, Steven Dirkse, Youngdae Kim, and Michael Ferris



Major Iteration Log
major minor  func  grad  residual    step  type prox    inorm  (label)
    0     0     1     1 0.0000e+00           I 0.0e+00 0.0e+00 (F_A)

Major Iterations. . . . 0
Minor Iterations. . . . 0
Restarts. . . . . . . . 0
Crash Iterations. . . . 0
Gradient Steps. . . . . 0
Function Evaluations. . 1
Gradient Evaluations. . 1
Basis Time. . . . . . . 0.000000
Total Time. . . . . . . 0.094000
Residual. . . . . . . . 0.000000e+00


Path 5.0.03 (Fri Jun 26 10:05:33 2020)
Written by Todd Munson, Steven Dirkse, Youngdae Kim, and Michael Ferris
Preprocessed size   : 19

Crash Log
major  func  diff  size  residual    step       prox   (label)
    0     0             0.0000e+00             0.0e+00 (F_A)

Major Iteration Log
major minor  func  grad  residual    step  type prox    inorm  (label)
    0     0     1     1 0.0000e+00           I 0.0e+00 0.0e+00 (F_A)

Major Iterations. . . . 0
Minor Iterations. . . . 0
Restarts. . . . . . . . 0
Crash Iterations. . . . 0
Gradient Steps. . . . . 0
Function Evaluations. . 1
Gradient Evaluations. . 1
Basis Time. . . . . . . 0.000000
Total Time. . . . . . . 0.015000
Residual. . . . . . . . 0.000000e+00
Postsolved residual: 0.0000e+00


MPSGE model with 3 sectors, 5 commodities and 1 consumers.
  Sectors: A (bm=1.0), B (bm=1.0), W (bm=1.0)
  Commodities: PX (bm=1.0), PY (bm=1.0), PW (bm=1.0), PL (bm=1.0), PK (bm=1.0)
  Consumers: CONS (bm=200.0)
Solution:
  A:	1.0
  B:	1.0
  W:	1.0
  PX:	1.0
  PY:	1.0
  PW:	1.0
  PL:	1.0
  PK:	1.0
  PX‡A:	80.0
  PY‡A:	20.0
  PX‡B:	20.0
  PY‡B:	80.0
  PW‡W:	200.0
  PL†A:	40.0
  PK†A:	60.0
  PL†B:	60.0
  PK†B:	40.0
  PX†W:	100.0
  PY†W:	100.0
  CONS:	200.0
  PWρCONS:	200.0


In [3]:
set_value(diff, 10.)
solve!(m)

θWiPX: 0.5=>0.55
Path 5.0.03 (Fri Jun 26 10:05:33 2020)
Written by Todd Munson, Steven Dirkse, Youngdae Kim, and Michael Ferris
Preprocessed size   : 19

Crash Log
major  func  diff  size  residual    step       prox   (label)
    0     0             1.9074e+01             0.0e+00 (F_PX)
    1     1     0    19 8.3461e-02  1.0e+00    0.0e+00 (F_PK†B)
pn_search terminated: no basis change.

Major Iteration Log
major minor  func  grad  residual    step  type prox    inorm  (label)
    0     0     2     2 8.3461e-02           I 0.0e+00 4.6e-02 (F_PK†B)
    1     1     3     3 1.7757e-05  1.0e+00 SO 0.0e+00 1.4e-05 (F_PK†B)
    2     1     4     4 1.4020e-12  1.0e+00 SO 0.0e+00 8.4e-13 (F_B)

Major Iterations. . . . 2
Minor Iterations. . . . 2
Restarts. . . . . . . . 0
Crash Iterations. . . . 1
Gradient Steps. . . . . 0
Function Evaluations. . 4
Gradient Evaluations. . 4
Basis Time. . . . . . . 0.000000
Total Time. . . . . . . 0.016000
Residual. . . . . . . . 1.402008e-12
Postsolved residu

MPSGE model with 3 sectors, 5 commodities and 1 consumers.
  Sectors: A (bm=1.0), B (bm=1.0), W (bm=1.0)
  Commodities: PX (bm=1.0), PY (bm=1.0), PW (bm=1.0), PL (bm=1.0), PK (bm=1.0)
  Consumers: CONS (bm=200.0)
Solution:
  A:	1.0767110404712248
  B:	0.9231469668931003
  W:	0.9523813418151985
  PX:	0.9552650300802475
  PY:	0.9493608751517101
  PW:	1.0
  PL:	0.9435734402066207
  PK:	0.9611892434237819
  PX‡A:	80.00000000000006
  PY‡A:	20.000000000000014
  PX‡B:	20.000000000000096
  PY‡B:	80.00000000000038
  PW‡W:	200.0
  PL†A:	40.67021653922481
  PK†A:	59.33595716535064
  PL†B:	60.889469228326895
  PK†B:	39.11871144924754
  PX†W:	109.8297687942915
  PY†W:	100.15523611484404
  CONS:	200.0
  PWρCONS:	200.0


MPSGE.jl  with s = 1.5, 2, 0.5, t both =0

In [157]:
var"A" = 1.07671104047122 ;
var"B" = 0.9231469668931 ;
var"W" = 0.952381341815198 ;
var"PX" = 0.955265030080247 ;
var"PY" = 0.94936087515171 ;
var"PW" = 1 ;
var"PL" = 0.94357344020662 ;
var"PK" = 0.961189243423781 ;
var"PX‡A" = 80 ;
var"PY‡A" = 20 ;
var"PX‡B" = 20 ;
var"PY‡B" = 80.0000000000003 ;
var"PW‡W" = 200 ;
var"PL†A" = 40.6702165392248 ;
var"PK†A" = 59.3359571653506 ;
var"PL†B" = 60.8894692283268 ;
var"PK†B" = 39.1187114492475 ;
var"PX†W" = 109.829768794291 ;
var"PY†W" = 100.155236114844 ;
var"CONS" = 200 ;
var"PWρCONS" = 200 ;
var"diff" = 10 ;
var"sub_elas_a" = 1.5 ;
var"sub_elas_b" = 2 ;
var"sub_elas_w" = 0.5 ;
var"t_elas_a" = 0.0
var"t_elas_b" = 0.0

0.0

GAMS with s = 1.5, 2, 0.5, t both =0

In [145]:
var"A" = 1.07671104 ;
var"B" = 0.92314697 ;
var"W" = 0.95231105 ;
var"PX" = 0.95519453 ;
var"PY" = 0.94929081 ;
var"PW" = 1 ;
var"PL" = 0.9435038 ;
var"PK" = 0.96111831 ;
var"PX‡A" = 80 ;
var"PY‡A" = 20 ;
var"PX‡B" = 20 ;
var"PY‡B" = 80 ;
var"PW‡W" = 200 ;
var"PL†A" = 40.67021654 ;
var"PK†A" = 59.33595717 ;
var"PL†B" = 60.88946923 ;
var"PK†B" = 39.11871145 ;
var"PX†W" = 109.8379 ;
var"PY†W" = 100.1626 ;
var"CONS" = 200 ;
var"PWρCONS" = 200 ;
var"diff" = 10 ;
var"sub_elas_a" = 1.5 ;
var"sub_elas_b" = 2 ;
var"sub_elas_w" = 0.5 ;


Compensated Demand
<br>PL†A

In [113]:
40.0 * (0.4 * (var"PL†A" / 40.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)) + 0.6 * (var"PK†A" / 60.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) ^ (1.0 / ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) * ((((0.4 * (var"PL" / 1.0) ^ (1.0 - var"sub_elas_a" * 1.0) + 0.6 * (var"PK" / 1.0) ^ (1.0 - var"sub_elas_a" * 1.0)) ^ (1.0 / (1.0 - var"sub_elas_a" * 1.0)) * (0.4 * (var"PL†A" / 40.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)) + 0.6 * (var"PK†A" / 60.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) ^ (1.0 / ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)))) * 1.0) / var"PL") ^ (var"sub_elas_a" * 1.0) - var"PL†A"

2.6706171496471143e-7

PK†A

In [114]:
60.0 * (0.4 * (var"PL†A" / 40.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)) + 0.6 * (var"PK†A" / 60.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) ^ (1.0 / ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) * ((((0.4 * (var"PL" / 1.0) ^ (1.0 - var"sub_elas_a" * 1.0) + 0.6 * (var"PK" / 1.0) ^ (1.0 - var"sub_elas_a" * 1.0)) ^ (1.0 / (1.0 - var"sub_elas_a" * 1.0)) * (0.4 * (var"PL†A" / 40.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)) + 0.6 * (var"PK†A" / 60.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) ^ (1.0 / ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)))) * 1.0) / var"PK") ^ (var"sub_elas_a" * 1.0) - var"PK†A"

-2.5405210379858545e-7

PL†B

In [115]:
60.0 * (0.6 * (var"PL†B" / 60.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)) + 0.4 * (var"PK†B" / 40.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) ^ (1.0 / ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) * ((((0.6 * (var"PL" / 1.0) ^ (1.0 - var"sub_elas_b" * 1.0) + 0.4 * (var"PK" / 1.0) ^ (1.0 - var"sub_elas_b" * 1.0)) ^ (1.0 / (1.0 - var"sub_elas_b" * 1.0)) * (0.6 * (var"PL†B" / 60.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)) + 0.4 * (var"PK†B" / 40.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) ^ (1.0 / ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)))) * 1.0) / var"PL") ^ (var"sub_elas_b" * 1.0) - var"PL†B"

3.49229146934249e-7

PK†B

In [116]:
40.0 * (0.6 * (var"PL†B" / 60.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)) + 0.4 * (var"PK†B" / 40.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) ^ (1.0 / ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) * ((((0.6 * (var"PL" / 1.0) ^ (1.0 - var"sub_elas_b" * 1.0) + 0.4 * (var"PK" / 1.0) ^ (1.0 - var"sub_elas_b" * 1.0)) ^ (1.0 / (1.0 - var"sub_elas_b" * 1.0)) * (0.6 * (var"PL†B" / 60.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)) + 0.4 * (var"PK†B" / 40.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) ^ (1.0 / ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)))) * 1.0) / var"PK") ^ (var"sub_elas_b" * 1.0) - var"PK†B"

-3.3803848253910473e-7

PX†W: 109.82976879429108 vs 109.8739

In [156]:
# "input.quantity"
(100.0 + var"diff") *
 (0.55
# "y_over_y_bar"
  # "y: theta"
# (0.52380952381 
    # (0.5
    # "y:get_comp_demand_name" #should this ALWAYS be DEMAND???
 * ((var"PX†W"
# + var"diff"
) 
  # "y:i.quantity"
/ (100.0 
+ var"diff"
)) 
  # "y: ^ρ" 
^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) +
  # "y: theta" 
# 0.47619047619 
0.5
  # "y:get_comp_demand_name"
 * ((var"PY†W"
# + var"diff"
  # "y:i.quantity"
) / (100.0
# + var"diff"
))
  # "y: ^ρ" 
^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))
  # "y: ^1/ρ"
) ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) *
# "create_cost_expr"
((((
  # "cost: theta"
  # 0.5
  0.55
  # 0.52380952381 
  # "cost: get_jump_variable_for_commodity(jm, input.commodity)"
 * (var"PX" 
  # "cost: benchmark(input.commod)"
 / 1.0) 
  # "cost: 1-elas"
 ^ (1.0 - var"sub_elas_w" * 1.0) 
+ 
  # "cost: theta"
0.5
# 0.47619047619
  # "cost: get_jump_variable_for_commodity(jm, input.commodity)"
 * (var"PY" 
  # "cost: benchmark(input.commod)"
/ 1.0) 
  # "cost: 1-elas"
^ (1.0 - var"sub_elas_w" * 1.0)) 
  # "cost:  1/1-elas"
^ (1.0 / (1.0 - var"sub_elas_w" * 1.0)) 
* (
  # "cost->y_over_y_bar"
  # "cost->y: theta"
# 0.52380952381 
    # 0.5
    0.55
    # "cost->y:get_comp_demand_name" #should this ALWAYS be DEMAND???
 * ((var"PX†W"
# + var"diff"
  # "cost->y:i.quantity"
) / (100.0
 + var"diff"
 )) 
  #  "cost->y: ^ρ"
 ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) + 
  #  "cost->y: theta"
# 0.47619047619
 0.5
  # "cost->y:get_comp_demand_name" #should this ALWAYS be DEMAND???
  * ((var"PY†W"
 # + var"diff"
  # "cost->y:i.quantity"
 ) / (100.0
# + var"diff"
 )) 
  # "cost->y: ^ρ"
 ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) 
    # "cost->y: ^1/ρ"
 ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)))) 
#  "get_commodity_benchmark"
 * 1.0) 
#  "get_jump_variable_for_commodity"
 / var"PX") ^
#  "s.elas"
 (var"sub_elas_w" * 1.0) #- var"PX†W"

107.19078780407948

With 0.5 for both share params: with MPSGE.jl values set, 109.82976879429108 as matches the results. With GAMS values set, 109.84192386634355 <br>
With 0.55 for what I believe to be ΘWiPX: with MPSGE.jl values set, 107.17892415325743. With the GAMS values set, 107.19078780407948<br>
With ΘWiPX=0.52380952381 and ΘWiPY=0.47619047619: with MPSGE.jl values set, 109.82571589469462. With the GAMS values set <b><u>109.83787251928246</u></b> as matches the GAMS result!


In [118]:
(100.0 + var"diff") * (0.5  * (var"PX†W"   / (100.0 + var"diff"  ))    ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) +    0.5    * ((var"PY†W"  ) / (100.0    ))   ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))   ) ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)))   * 
((((0.5
 * (var"PX" / 1.0) ^ (1.0 - var"sub_elas_w" * 1.0) + 0.5 * (var"PY" / 1.0) ^ (1.0 - var"sub_elas_w" * 1.0)) ^ (1.0 / (1.0 - var"sub_elas_w" * 1.0)) * (0.5 * (var"PX†W" / (100.0 + var"diff")) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) + 0.5 * (var"PY†W" / 100.0) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)))) * 1.0) / var"PX") ^ (var"sub_elas_w" * 1.0) - var"PX†W"

0.004023866343544569

In [119]:
var"PX†W"

109.8379

  ┴  PX†W

In [120]:
(100.0 + var"diff") *
 (0.5 * (var"PX†W" / (100.0 + var"diff")) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) + 0.5 * (var"PY†W" / 100.0) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) ^ (1.0 / ((sub_elas_w * 1.0 - 1.0) / (sub_elas_w * 1.0))) * 
((((0.5 * (PX / 1.0) ^ (1.0 - var"sub_elas_w" * 1.0) + 0.5 * (var"PY" / 1.0) ^ (1.0 - var"sub_elas_w" * 1.0)) ^ (1.0 / (1.0 - var"sub_elas_w" * 1.0)) * (0.5 * (var"PX†W" / (100.0 + var"diff")) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) + 0.5 * (var"PY†W" / 100.0) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)))) * 1.0) / PX) ^ (var"sub_elas_w" * 1.0) - var"PX†W"

0.004023866343544569

PY†W: 100.15523611484365 vs 100.1626

In [127]:
100.0 * (
  # 0.52380952381 # The value suggested by the ipynb 110/210
  # .55 # The value as updated now (1/30/23), 110/200
  0.5  #unchanged from diff = 0
   * ((var"PX†W"
#  + var"diff"
)/ (100.0
 + var"diff"
 )) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) + 
 0.5 #unchanged
#  0.47619047619 # The value suggested by the ipynb 100/210
  * (var"PY†W" / 100.0) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) * ((((
    # 0.52380952381 # The value suggested by the ipynb 110/210 
    0.55 # The value as updated now (1/30/23), 110/200
    # 0.5  #unchanged from diff = 0
     * (var"PX" / 1.0) ^ (1.0 - var"sub_elas_w" * 1.0) + 
  0.5 #unchanged
  # 0.47619047619 # The value suggested by the ipynb 100/210
  * (var"PY" / 1.0) ^ (1.0 - var"sub_elas_w" * 1.0)) ^ (1.0 / (1.0 - var"sub_elas_w" * 1.0)) *(
  # 0.55 # The value as updated now (1/30/23), 110/200
  0.5
  # 0.52380952381 # The value suggested by the ipynb 110/210 
  * ((var"PX†W"
# + var"diff"
)/ (100.0
  + var"diff"
  )) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) + 
  0.5  #unchanged
  # 0.47619047619 # The value suggested by the ipynb 100/210 
   * (var"PY†W" / 100.0) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)))) * 1.0) / var"PY") ^ (var"sub_elas_w" * 1.0) #- var"PY†W"

105.18239922267419

Compensated Supply
<br>PX‡A: 79.99999999999976

In [142]:
80.0 * (0.4 * (var"PL†A" / 40.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)) + 0.6 * (var"PK†A" / 60.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) ^ (1.0 / ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) * (var"PX" / ((((80.0 * 1.0) / (80.0 * 1.0 + 20.0 * 1.0)) * (var"PX" / 1.0) ^ (1.0 + var"t_elas_a" * 1.0) + ((20.0 * 1.0) / (80.0 * 1.0 + 20.0 * 1.0)) * (var"PY" / 1.0) ^ (1.0 + var"t_elas_a" * 1.0)) ^ (1.0 / (1.0 + var"t_elas_a" * 1.0)) * 1.0)) ^ (var"t_elas_a" * 1.0) #- var"PX‡A"

80.0000000043603

PY‡A

In [143]:
20.0 * (0.4 * (var"PL†A" / 40.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0)) + 0.6 * (var"PK†A" / 60.0) ^ ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) ^ (1.0 / ((var"sub_elas_a" * 1.0 - 1.0) / (var"sub_elas_a" * 1.0))) * (var"PY" / ((((80.0 * 1.0) / (80.0 * 1.0 + 20.0 * 1.0)) * (var"PX" / 1.0) ^ (1.0 + var"t_elas_a" * 1.0) + ((20.0 * 1.0) / (80.0 * 1.0 + 20.0 * 1.0)) * (var"PY" / 1.0) ^ (1.0 + var"t_elas_a" * 1.0)) ^ (1.0 / (1.0 + var"t_elas_a" * 1.0)) * 1.0)) ^ (var"t_elas_a" * 1.0) - var"PY‡A"

1.090075585352679e-9

PX‡B

In [144]:
20.0 * (0.6 * (var"PL†B" / 60.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)) + 0.4 * (var"PK†B" / 40.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) ^ (1.0 / ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) * (var"PX" / ((((20.0 * 1.0) / (20.0 * 1.0 + 80.0 * 1.0)) * (var"PX" / 1.0) ^ (1.0 + var"t_elas_b" * 1.0) + ((80.0 * 1.0) / (20.0 * 1.0 + 80.0 * 1.0)) * (var"PY" / 1.0) ^ (1.0 + var"t_elas_b" * 1.0)) ^ (1.0 / (1.0 + var"t_elas_b" * 1.0)) * 1.0)) ^ (var"t_elas_b" * 1.0) - var"PX‡B"

4.843947465360543e-10

PY‡B

In [145]:
80.0 * (0.6 * (var"PL†B" / 60.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0)) + 0.4 * (var"PK†B" / 40.0) ^ ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) ^ (1.0 / ((var"sub_elas_b" * 1.0 - 1.0) / (var"sub_elas_b" * 1.0))) * (var"PY" / ((((20.0 * 1.0) / (20.0 * 1.0 + 80.0 * 1.0)) * (var"PX" / 1.0) ^ (1.0 + var"t_elas_b" * 1.0) + ((80.0 * 1.0) / (20.0 * 1.0 + 80.0 * 1.0)) * (var"PY" / 1.0) ^ (1.0 + var"t_elas_b" * 1.0)) ^ (1.0 / (1.0 + var"t_elas_b" * 1.0)) * 1.0)) ^ (var"t_elas_b" * 1.0) - var"PY‡B"

1.937578986144217e-9

PW‡W: 199.99999999999946, with GAMS, the subtraction equation = 0.014756012951039565

In [182]:
200.0 * (0.5 * (var"PX†W" / (100.0 + var"diff")) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0)) + 0.5 * (var"PY†W" / 100.0) ^ ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) ^ (1.0 / ((var"sub_elas_w" * 1.0 - 1.0) / (var"sub_elas_w" * 1.0))) * (var"PW" / ((+(((200.0 * 1.0) / +(200.0 * 1.0)) * (var"PW" / 1.0) ^ (1.0 + 0.0))) ^ (1.0 / (1.0 + 0.0)) * 1.0)) ^ 0.0 - var"PW‡W"

0.014756012951039565

Final Demand
<br> PW = 1, CONS=200

In [148]:
200.0 * (var"CONS" / 200.0) * (1.0 / var"PW") - var"PWρCONS"


0.0

In [149]:
(var"PL" * var"PL†A" + var"PK" * var"PK†A") - (var"PX" * var"PX‡A" + var"PY" * var"PY‡A")


1.2977564267657726e-7

In [150]:
(var"PL" * var"PL†B" + var"PK" * var"PK†B") - (var"PX" * var"PX‡B" + var"PY" * var"PY‡B")


3.6689726812255685e-8

In [151]:
(var"PX" * var"PX†W" + var"PY" * var"PY†W") - +(var"PW" * var"PW‡W")


-3.0476069809992623e-6

Market Clearance <br>
PY

In [152]:
(0.0 + var"A" * var"PY‡A" + var"B" * var"PY‡B") - (0.0 + var"W" * var"PY†W")


2.7623270000276534e-5

PX

In [153]:
# (0.0 + var"A" * var"PX‡A" + var"B" * var"PX‡B")
 - (0.0 + var"W") 
#  0.95231105* var"PX†W"
#  )


-0.95231105

In [155]:
(0.0 + var"A" * var"PX‡A" + var"B" * var"PX‡B") - (0.0 + var"W" * var"PX†W")


-2.327879501251573e-5

PW=1, PW‡W ~200, PWρCONS = 200, so why is W wrong?

In [156]:
(0.0 + var"W" * var"PW‡W") - (0.0 + var"PWρCONS")


-9.537790000000001

In [157]:
(0.0 + 0.95231105 * var"PW‡W") - (0.0 + var"PWρCONS")


-9.537790000000001

PL

In [158]:
(9.523731636960406-9.537790000000001)+0.9523813418151985

0.9383229787756033

In [159]:
(0.0 + 100.0) - (0.0 + var"A" * var"PL†A" + var"B" * var"PL†B")


-1.7239133853763633e-7

PK

In [160]:
(0.0 + 100.0) - (0.0 + var"A" * var"PK†A" + var"B" * var"PK†B")


-9.92779689568124e-8

Income Balance:
<br>CONS

In [169]:
((100) * var"PL" + (100.0) * var"PK") - var"CONS"


-9.537788999999975

In [171]:
var"PK"

0.96111831

In [ ]:
m = Model()

@parameter(m, endow, 1.0)
@parameter(m, diff, 0.0)

@sector(m, X)
@sector(m, Y)
@sector(m, U)

@commodity(m, PX)
@commodity(m, PY)
@commodity(m, PU)
@commodity(m, PL)
@commodity(m, PK)

@consumer(m, RA, benchmark = 150.)

@production(m, X, 0, 1, [Output(PX, 100)], [Input(PL, :(50+$diff)), Input(PK, 50)])
@production(m, Y, 0, 1, [Output(PY, 50)], [Input(PL, 20), Input(PK, 30)])
@production(m, U, 0, 1, [Output(PU, 150)], [Input(PX, 100), Input(PY, 50)]) 

@demand(m, RA, 1.0, [Demand(PU, 150)], [Endowment(PL, :(70 * $endow)), Endowment(PK, 80.)])
solve!(m)

sub_elas_1


sub_elas_1
sub_elas_1


sub_elas_1
sub_elas_1
sub_elas_1


Path 5.0.03 (Fri Jun 26 10:05:33 2020)
Written by Todd Munson, Steven Dirkse, Youngdae Kim, and Michael Ferris

Crash Log
major  func  diff  size  residual    step       prox   (label)


    0     0             0.0000e+00             0.0e+00 (F_X)

Major Iteration Log
major minor  func  grad  residual    step  type prox    inorm  (label)
    0     0     1     1 0.0000e+00           I 0.0e+00 0.0e+00 (F_X)

Major Iterations. . . . 0


Minor Iterations. . . . 0
Restarts. . . . . . . . 0
Crash Iterations. . . . 0
Gradient Steps. . . . . 0
Function Evaluations. . 1
Gradient Evaluations. . 1


Basis Time. . . . . . . 0.000000
Total Time. . . . . . . 0.140000
Residual. . . . . . . . 0.000000e+00


MPSGE model with 3 sectors, 5 commodities and 1 consumers.
  Sectors: X (bm=1.0), Y (bm=1.0), U (bm=1.0)
  Commodities: PX (bm=1.0), PY (bm=1.0), PU (bm=1.0), PL (bm=1.0), PK (bm=1.0)
  Consumers: RA (bm=150.0)
Solution:
  X:	1.0
  Y:	1.0
  U:	1.0
  PX:	1.0
  PY:	1.0
  PU:	1.0
  PL:	1.0
  PK:	1.0
  PX‡X:	100.0
  PY‡Y:	50.0
  PU‡U:	150.0
  PL†X:	50.0
  PK†X:	50.0
  PL†Y:	20.0
  PK†Y:	30.0
  PX†U:	100.0
  PY†U:	50.0
  RA:	150.0
  PUρRA:	150.0


In [32]:
algebraic_version(m)

Mixed complementarity problem with 19 constraints:
  Compensated Demand
    (50.0 + diff) * ((PL†X / (50.0 + diff)) ^ (((50.0 + diff) * 1.0) / +(100.0 * 1.0)) * (PK†X / 50.0) ^ ((50.0 * 1.0) / +(100.0 * 1.0))) * (((((PL / 1.0) ^ 0.5 * (PK / 1.0) ^ 0.5) * ((PL†X / (50.0 + diff)) ^ 0.5 * (PK†X / 50.0) ^ 0.5)) * 1.0) / PL) ^ 1.0 - PL†X                              ┴  PL†X
    50.0 * ((PL†X / (50.0 + diff)) ^ (((50.0 + diff) * 1.0) / +(100.0 * 1.0)) * (PK†X / 50.0) ^ ((50.0 * 1.0) / +(100.0 * 1.0))) * (((((PL / 1.0) ^ 0.5 * (PK / 1.0) ^ 0.5) * ((PL†X / (50.0 + diff)) ^ 0.5 * (PK†X / 50.0) ^ 0.5)) * 1.0) / PK) ^ 1.0 - PK†X                                       ┴  PK†X
    20.0 * ((PL†Y / 20.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (PK†Y / 30.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0))) * (((((PL / 1.0) ^ 0.4 * (PK / 1.0) ^ 0.6) * ((PL†Y / 20.0) ^ 0.4 * (PK†Y / 30.0) ^ 0.6)) * 1.0) / PL) ^ 1.0 - PL†Y                                                                    ┴  PL†Y
    30.0 * ((PL†Y / 20.0) ^ 

In [33]:
set_value(diff, 10.0)
set_fixed!(RA, true)
solve!(m)

MethodError: MethodError: no method matching set_value(::Int64, ::Float64)
Closest candidates are:
  set_value(!Matched::JuMP.NonlinearParameter, ::Number) at C:\Users\Eli\.julia\packages\JuMP\zn6NT\src\nlp.jl:154
  set_value(!Matched::MPSGE.CommodityRef, ::Float64) at c:\Users\Eli\.julia\dev\MPSGE\src\model.jl:661
  set_value(!Matched::MPSGE.AuxRef, ::Float64) at c:\Users\Eli\.julia\dev\MPSGE\src\model.jl:651
  ...

copy and paste results into Excel to turn into  var"Variable" = Value;

Any variable that is given a value can no longer be updated with set_value


This WAS MPSGE.jl + DIFF IN INPUT-> PL of X: from updating diff = 10 and set_fixed(RA, true), not match to GAMS

In [34]:
var"X" = 0.907085116015307;
var"Y" = 1.0099816812365;
var"U" = 0.940163081141929;
var"PX" = 1.10243237351448;
var"PY" = 0.990116965766566;
var"PU" = 1.06364525277147;
var"PL" = 1.06493506229893;
var"PK" = 0.943181817041456;
var"PX‡X" = 100.000000061341;
var"PY‡Y" = 50.0000000276231;
var"PU‡U" = 150.000000000967;
var"PL†X" = 56.4660485536119;
var"PK†X" = 53.1292711343664;
var"PL†Y" = 18.5948796353724;
var"PK†Y" = 31.4928770648795;
var"PX†U" = 96.4816779936831;
var"PY†U" = 53.7131111648588;
var"RA" = 150;
var"PUρRA" = 141.024462172411;
var"diff" = 10 ;
var"endow" = 1;

GAMS + DIFF IN INPUT -> PL of X: from updating diff=10, not RA.FX = RA.FX

In [35]:
var"X" = 0.90209303;
var"Y" = 1.00998168;
var"U" = 0.93671049;
var"PX" = 1.10853311;
var"PY" = 0.99011696;
var"PU" = 1.06756571;
var"PL" = 1.06493506;
var"PK" = 0.94318182;
var"PX‡X" = 100;
var"PY‡Y" = 50;
var"PU‡U" = 150;
var"PL†X" = 56.77852579;
var"PK†X" = 53.42328265;
var"PL†Y" = 18.59487971;
var"PK†Y" = 31.49287695;
var"PX†U" = 96.30435869;
var"PY†U" = 53.91109057;
var"RA" = 150;
var"PUρRA" = 140.5066;
var"diff" = 10 ;
var"endow" = 1;

This is MPSGE.jl + diff in output of X, from updating diff = 10 and set_fixed(RA, true), ~ match to GAMS

In [36]:
var"X" = 1.0000000000544 ;
var"Y" = 1.00000000000066 ;
var"U" = 1.0656022367644 ;
var"PX" = 0.909090908368919 ;
var"PY" = 0.999999999222552 ;
var"PU" = 0.938436467833448 ;
var"PL" = 0.999999999157994 ;
var"PK" = 0.999999999269399 ;
var"PX‡X" = 109.999999994235 ;
var"PY‡Y" = 50.0000000001482 ;
var"PU‡U" = 150.000000000481 ;
var"PL†X" = 49.9999999971192 ;
var"PK†X" = 49.9999999971148 ;
var"PL†Y" = 20.0000000001045 ;
var"PK†Y" = 30.0000000000426 ;
var"PX†U" = 103.228011545968 ;
var"PY†U" = 46.921823429979 ;
var" RA" = 150 ;
var"PUρRA" = 159.840335515262 ;
var"diff" = 10 ;


GAMS for diff=10, RA.FX = 150, SAME as no RA.FX

In [37]:
var"X" = 1 ;
var"Y" = 1 ;
var"U" = 1.06560224 ;
var"PX" = 0.90909091 ;
var"PY" = 1 ;
var"PU" = 0.93843647 ;
var"PL" = 1 ;
var"PK" = 1 ;
var"PX‡X" = 110 ;
var"PY‡Y" = 50 ;
var"PU‡U" = 150 ;
var"PL†X" = 50 ;
var"PK†X" = 50 ;
var"PL†Y" = 20 ;
var"PK†Y" = 30 ;
var"PX†U" = 103.228 ;
var"PY†U" = 46.92182343 ;
var" RA" = 150 ;
var"PUρRA" = 159.8403 ;
var"diff" = 10 ;

In [38]:
set_value(diff, 10.0)
set_value(endow, 1.1)
set_fixed!(RA, true)
solve!(m)

MethodError: MethodError: no method matching set_value(::Int64, ::Float64)
Closest candidates are:
  set_value(!Matched::JuMP.NonlinearParameter, ::Number) at C:\Users\Eli\.julia\packages\JuMP\zn6NT\src\nlp.jl:154
  set_value(!Matched::MPSGE.CommodityRef, ::Float64) at c:\Users\Eli\.julia\dev\MPSGE\src\model.jl:661
  set_value(!Matched::MPSGE.AuxRef, ::Float64) at c:\Users\Eli\.julia\dev\MPSGE\src\model.jl:651
  ...

set_value(diff, 10.0)
set_value(endow, 1.1)
set_fixed!(RA, true)

In [39]:
var"X" = 1.04904713847483 ;
var"Y" = 1.03886011792587 ;
var"U" = 1.11408336425776 ;
var"PX" = 0.907215362082662 ;
var"PY" = 1.00751453099342 ;
var"PU" = 0.939486844766213 ;
var"PL" = 0.951515148888152 ;
var"PK" = 1.04666666566031 ;
var"PX‡X" = 109.977284836133 ;
var"PY‡Y" = 50.0000000174636 ;
var"PU‡U" = 150.000000000728 ;
var"PL†X" = 52.4285306016761 ;
var"PK†X" = 47.6623005410971 ;
var"PL†Y" = 21.177057063675 ;
var"PK†Y" = 28.8778050917323 ;
var"PX†U" = 103.557201960118 ;
var"PY†U" = 46.6239848677298 ;
var" RA" = 157 ;
var"PUρRA" = 167.112504639448 ;
var"diff" = 10 ;

GAMS diff=10, endow=1.1, RA.FX=80+70*1.1 SAME as NO RA.FX

In [40]:
var"X" = 1.04880885 ;
var"Y" = 1.03886012 ;
var"U" = 1.11406803 ;
var"PX" = 0.9072341 ;
var"PY" = 1.00751453 ;
var"PU" = 0.93949978 ;
var"PL" = 0.95151515 ;
var"PK" = 1.04666667 ;
var"PX‡X" = 110 ;
var"PY‡Y" = 50 ;
var"PU‡U" = 150 ;
var"PL†X" = 52.44044241 ;
var"PK†X" = 47.67312946 ;
var"PL†Y" = 21.17705706 ;
var"PK†Y" = 28.87780508 ;
var"PX†U" = 103.5565 ;
var"PY†U" = 46.6246268 ;
var" RA" = 157 ;
var"PUρRA" = 167.1102 ;
# var"diff" = 10 ;
var"diff" = 0 ;

GAMS for diff=10, PX.FX=1

In [41]:
var"X" = 1 ;
var"Y" = 1 ;
var"U" = 1.06560224 ;
var"PX" = 1 ;
var"PY" = 1.1 ;
var"PU" = 1.03228012 ;
var"PL" = 1.1 ;
var"PK" = 1.1 ;
var"PX‡X" = 110 ;
var"PY‡Y" = 50 ;
var"PU‡U" = 150 ;
var"PL†X" = 50 ;
var"PK†X" = 50 ;
var"PL†Y" = 20 ;
var"PK†Y" = 30 ;
var"PX†U" = 103.228 ;
var"PY†U" = 46.92182343 ;
var" RA" = 165 ;
var"PUρRA" = 159.840335515262 ;
var"diff" = 10 ;

In [42]:
var"X" = 1.04904713847483 ;
var"Y" = 1.03886011792587 ;
var"U" = 1.11408336425776 ;
var"PX" = 0.907215362082662 ;
var"PY" = 1.00751453099342 ;
var"PU" = 0.939486844766213 ;
var"PL" = 0.951515148888152 ;
var"PK" = 1.04666666566031 ;
var"PX‡X" = 109.977284836133 ;
var"PY‡Y" = 50.0000000174636 ;
var"PU‡U" = 150.000000000728 ;
var"PL†X" = 52.4285306016761 ;
var"PK†X" = 47.6623005410971 ;
var"PL†Y" = 21.177057063675 ;
var"PK†Y" = 28.8778050917323 ;
var"PX†U" = 103.557201960118 ;
var"PY†U" = 46.6239848677298 ;
var" RA" = 157 ;
var"PUρRA" = 167.112504639448 ;
var"diff" = 10 ;

set_value(diff, 10.0)
set_value(endow, 1.1)
set_fixed!(RA, false)

In [43]:
var"X" = 1.03523426518343 ;
var"Y" = 1.03886011829416 ;
var"U" = 1.11318877431167 ;
var"PX" = 1.02338957692563 ;
var"PY" = 1.1351639913029 ;
var"PU" = 1.05936802743385 ;
var"PL" = 1.07206963405921 ;
var"PK" = 1.1792765990823 ;
var"PX‡X" = 111.310479273694 ;
var"PY‡Y" = 50.0000000039924 ;
var"PU‡U" = 150.000000097099 ;
var"PL†X" = 53.1280714406104 ;
var"PK†X" = 48.2982467638607 ;
var"PL†Y" = 21.177057057939 ;
var"PK†Y" = 28.8778050837489 ;
var"PX†U" = 103.515616409909 ;
var"PY†U" = 46.6614531888854 ;
var" RA" = 176.891489749144 ;
var"PUρRA" = 166.978316299905 ;
var"diff" = 10;

 Compensated Demand


PL†X

In [44]:
(50.0+var"diff") * ((var"PL†X" / (50.0+var"diff")) ^ (((50.0) * 1.0) / +((100.0 ) * 1.0)) * (var"PK†X" / 50.0) ^ ((50.0 * 1.0) / +((100.0 ) * 1.0))) * (((((var"PL" / 1.0) ^ ((50.0 * 1.0) / +((100.0 ) * 1.0)) * (var"PK" / 1.0) ^ ((50.0 * 1.0) / +((100.0 ) * 1.0))) * ((var"PL†X" / (50.0+var"diff")) ^ (((50.0) * 1.0) / +((100.0 ) * 1.0)) * (var"PK†X" / 50.0) ^ ((50.0 * 1.0) / +((100.0 ) * 1.0)))) * 1.0) / PL) ^ 1.0 - var"PL†X"

0.6966456532467049

With GAMS values <br>
ABOVE: Adding +diff to *1* of the numerators in the exponent of (PL:X/50+diff) BUT NOT the other one! -4.185437774140155e-9 , OR ALSO to the denominators of that exponent and the adjascent PK:X, -4.515321450071497e-9

PK†X

In [45]:
(50.0) * ((var"PL†X" / (50.0+var"diff")) ^ (((50.0+var"diff") * 1.0) / +((100.0+var"diff" ) * 1.0)) * (var"PK†X" / 50.0) ^ (((50.0) * 1.0) / +((100.0+var"diff" ) * 1.0))) * (((((var"PL" / 1.0) ^ ((50.0 * 1.0) / +((100.0 ) * 1.0)) * (var"PK" / 1.0) ^ ((50.0 * 1.0) / +((100.0 ) * 1.0))) * ((var"PL†X" / (50.0+var"diff")) ^ (((50.0) * 1.0) / +((100.0 ) * 1.0)) * (var"PK†X" / 50.0) ^ (((50.0) * 1.0) / +((100.0 ) * 1.0)))) * 1.0) / PK) ^ 1.0 - var"PK†X"

-7.682900310833432

PL†Y

In [46]:
20.0 * ((var"PL†Y" / 20.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (var"PK†Y" / 30.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0))) * (((((var"PL" / 1.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (var"PK" / 1.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0))) * ((var"PL†Y" / 20.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (var"PK†Y" / 30.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0)))) * 1.0) / PL) ^ 1.0 - var"PL†Y"

2.1099900493481982e-8

PK†Y

In [47]:
30.0 * ((var"PL†Y" / 20.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (var"PK†Y" / 30.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0))) * (((((var"PL" / 1.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (var"PK" / 1.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0))) * ((var"PL†Y" / 20.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (var"PK†Y" / 30.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0)))) * 1.0) / PK) ^ 1.0 - var"PK†Y"

-1.5569384004265885e-8

PX†U

In [48]:
100.0 * ((var"PX†U" / 100.0) ^ ((100.0 * 1.0) / +(150.0 * 1.0)) * (var"PY†U" / 50.0) ^ ((50.0 * 1.0) / +(150.0 * 1.0))) * (((((var"PX" / 1.0) ^ ((100.0 * 1.0) / +(150.0 * 1.0)) * (var"PY" /1.0) ^ ((50.0 * 1.0) / +(150.0 * 1.0))) * ((var"PX†U" / 100.0) ^ ((100.0 * 1.0) / +(150.0 * 1.0)) * (var"PY†U" / 50.0) ^ ((50.0 * 1.0) / +(150.0 * 1.0)))) * 1.0) / PX) ^ 1.0 - var"PX†U"

6.712691913435265e-8

PY†U

In [49]:
50.0 * ((var"PX†U" / 100.0) ^ ((100.0 * 1.0) / +(150.0 * 1.0)) * (var"PY†U" / 50.0) ^ ((50.0 * 1.0) / +(150.0 * 1.0))) * (((((var"PX" / 1.0) ^ ((100.0 * 1.0) / +(150.0 * 1.0)) * (var"PY" /1.0) ^ ((50.0 * 1.0) / +(150.0 * 1.0))) * ((var"PX†U" / 100.0) ^ ((100.0 * 1.0) / +(150.0 * 1.0)) * (var"PY†U" / 50.0) ^ ((50.0 * 1.0) / +(150.0 * 1.0)))) * 1.0) / PY) ^ 1.0 - var"PY†U"

2.988659986158382e-8

  Compensated Supply


PX‡X

In [50]:
(100.0 ) * ((var"PL†X" / (50.0+var"diff")) ^ (((50.0+var"diff") * 1.0) / +((100.0+var"diff" ) * 1.0)) * (var"PK†X" / 50.0) ^ ((50.0 * 1.0) / +((100.0+var"diff" ) * 1.0))) * ((var"PX" / (+((((100.0 ) * 1.0) / +((100.0 ) * 1.0)) * (var"PX" / 1.0) ^ (1.0 + 0.0))) ^ (1.0 / (1.0 + 0.0))) / 1.0) ^ 0.0 - var"PX‡X" 

-19.191435875898804

Above: Added +var"diff" to the numerator of the exponenet of PL:X. Hypothesis is that it either SHOULD be there (or - less likely - that MPSGE is wrong)

PY‡Y

In [51]:
50.0 * ((var"PL†Y" / 20.0) ^ ((20.0 * 1.0) / +(50.0 * 1.0)) * (var"PK†Y" / 30.0) ^ ((30.0 * 1.0) / +(50.0 * 1.0))) * ((var"PY" /(+(((50.0 * 1.0) / +(50.0 * 1.0)) * (var"PY" /1.0) ^ (1.0 + 0.0))) ^ (1.0 / (1.0 + 0.0))) / 1.0) ^ 0.0 - var"PY‡Y"

-2.396873810539546e-10

PU‡U

In [52]:
150.0 * ((var"PX†U" / 100.0) ^ ((100.0 * 1.0) / +(150.0 * 1.0)) * (var"PY†U" / 50.0) ^ ((50.0 * 1.0) / +(150.0 * 1.0))) * ((var"PU" / (+(((150.0 * 1.0) / +(150.0 * 1.0)) * (var"PU" / 1.0) ^ (1.0 + 0.0))) ^ (1.0 / (1.0 + 0.0))) / 1.0) ^ 0.0 - var"PU‡U"

-2.269189280923456e-10

  Final Demand


In [53]:
150.0 * (
    # 165
    # 157 
var"RA"
 / 150.0) * (1.0 / var"PU") - var"PUρRA"

-25.384464007278154

  Zero Profit


In [54]:
(var"PL" * var"PL†X" + var"PK" * var"PK†X") - +(var"PX" * var"PX‡X")

1.6352998954971554e-9

In [55]:
(var"PL" * var"PL†Y" + var"PK" * var"PK†Y") - +(var"PY" * var"PY‡Y")

9.003599643619964e-9

In [56]:
(var"PX" * var"PX†U" + var"PY" * var"PY†U") - +(var"PU" * var"PU‡U")

1.0688012253012857e-7

  Market clearance


In [57]:
(0.0 + 70.0 * 
# 1.1
# 1.0
var"endow"
) - (0.0 + var"X" * var"PL†X" + var"Y" * var"PL†Y")

-6.999999998765759

In [58]:
(0.0 + 80.0) - (0.0 + var"X" * var"PK†X" + var"Y" * var"PK†Y")

-3.612385057749634e-9

In [59]:
(0.0 + var"U" * var"PU‡U") - (0.0 + var"PUρRA")

-4.506500772549771e-8

In [60]:
(0.0 + var"Y" * var"PY‡Y") - (0.0 + var"U" * var"PY†U")

3.591883768194748e-8

In [61]:
(0.0 + var"X" * var"PX‡X") - (0.0 + var"U" * var"PX†U")

6.465444357672823e-8

  Income balance


In [62]:
((70.0 *
# 1.0
# 1.1 
var"endow"
) * var"PL" + 80.0 * var"PK") - 
# 157
# 165
var"RA"

19.38700231072869

In [63]:
set_value(diff,10.)
set_fixed!(PX, true)
# set_value(endow, 1.1)
# set_value(RA,165.)
# set_fixed!(RA,true)
solve!(m)

MethodError: MethodError: no method matching set_value(::Int64, ::Float64)
Closest candidates are:
  set_value(!Matched::JuMP.NonlinearParameter, ::Number) at C:\Users\Eli\.julia\packages\JuMP\zn6NT\src\nlp.jl:154
  set_value(!Matched::MPSGE.CommodityRef, ::Float64) at c:\Users\Eli\.julia\dev\MPSGE\src\model.jl:661
  set_value(!Matched::MPSGE.AuxRef, ::Float64) at c:\Users\Eli\.julia\dev\MPSGE\src\model.jl:651
  ...